interface/forms/phq9/common.php

```diff
@@ -0,0 +1,340 @@
+            question.appendChild(text);
+            var new_line = document.createElement("br");
+            var ital = document.createElement("i"); // second part is in italics
+            var question_2 = document.createTextNode(jsAttr(<?php echo   js_escape($str_q10_2); ?>));
+            ital.appendChild(question_2);
+            question.name = "tenth";
+            question.appendChild(new_line);
+            question.appendChild(ital);
+// populate the   the menue
+            menue.options[0] = new Option( <?php echo js_escape($str_not); ?>, "0");
+            menue.options[1] = new Option( <?php echo js_escape($str_somewhat); ?>, "1");
+            menue.options[2] = new Option( <?php echo js_escape($str_very); ?>, "2");
+            menue.options[3] = new Option( <?php echo js_escape($str_extremely);?>, "3");
+            menue.options[4] = new Option( <?php echo js_escape($str_default);  ?>, "undef");
+        }
+    </script>
+    <div class="col-12">
+        <h3><?php echo text($str_form_name); ?></h3>
+        <form method=post action="<?php echo $rootdir; ?>/forms/phq9/save.php?mode=<?php echo attr_url($viewmode); ?>&id=<?php echo attr_url($_GET['id'] ?? 0); ?>" name="my_form"<?php if (!$obj) {
+            ?> onSubmit="return(check_all());"<?php }?>>
+            <input type="hidden" name="csrf_token_form" value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
+            <div class="title"><?php echo xlt('How often have you been bothered by the following over the past 2 weeks?'); ?></div>
+            <hr />
+            <table>
+                <tr>
+                    <td>
+                        <span class="label"><?php echo ++$qno . ". " . xlt('Little interest or pleasure in doing things'); ?></span>
+                        <select class="form-input" name="interest_score" onchange="update_score(0, my_form.interest_score.value);">
+                            <?php if (!$obj) {
+                                ?><?php if (!$obj) {
+    ?><option value="undef"><?php echo text($str_default); ?></option><?php } ?><?php } ?>
+                            <option value="0"><?php echo text($str_not); ?></option>
+                            <option value="1"><?php echo text($str_several); ?></option>
+                            <option value="2"><?php echo text($str_more); ?></option>
+                            <option value="3"><?php echo text($str_nearly); ?></option>
+                        </select>
+                        <?php if ($obj) { ?>
+                        <script>
+                            // set the default to the previous value - so it is displayed in the menue box
+                            document.my_form.interest_score.options[<?php echo text($obj['interest_score']); ?>].defaultSelected=true;
+                            var i = <?php echo text($obj['interest_score']); ?> ; //the value from last time
```

Comment:
js_escape() instead of text() in line above

Reply:
@bradymiller - in this case we want a numerical integer value, used to calculate the patient's score on this form. so text or attr - i'll change to attr - similarly at the other places where this occurs

Reply:
@ruthkonyn , note js_escape is using json_encode which should leave integers alone (ie.basically will leave them be which can check out here: https://onlinephp.io/json-encode ).

Reply:
@bradymiller the objective of these lines is to extract the score for the individual question and sum them into a total integer value. when i use js_escape and display the value of the score i get the ascii values, e.g. if i use js_escape to add, say, 1+4+1 the score displays as 141 instead of 6. It seems that using js_escape casts 'i' to be text array, rather than an integer.  there may be ways round this, s.a. casting i to be integer each time, but why are you recommending using js_escape rather than the simpler attr() or text() ? I don't see the need to get json values here (which are always strings?). thanks.